responsemanager/peerresponsemanager/peerresponsesender.go

```diff
@@ -383,6 +392,12 @@ func (prs *peerResponseSender) FinishWithCancel(requestID graphsync.RequestID) {
 }
 
 func (prs *peerResponseSender) buildResponse(blkSize uint64, buildResponseFn func(*responsebuilder.ResponseBuilder), notifees []notifications.Notifee) bool {
+	allocResponse := prs.allocator.AllocateBlockMemory(prs.p, blkSize)
+	select {
+	case <-prs.ctx.Done():
+		return false
```

Comment:
There are two cases this could happen -- first is if the whole graphsync instance shuts down, in which case, who cares. 

The second is if the peer itself shuts down. I am working on a fix for this now.